src/cargo/core/resolver/mod.rs

```diff
@@ -1116,27 +1219,44 @@ fn activate_deps_loop(
                 candidate.summary.version()
             );
             let res = activate(&mut cx, registry, Some(&parent), candidate, &method);
-            successfully_activated = res.is_ok();
 
-            match res {
+            let successfully_activated = match res {
+                // Success! We've now activated our `candidate` in our context
+                // and we're almost ready to move on. We may want to scrap this
+                // frame in the end if it looks like it's not going to end well,
+                // so figure that out here.
                 Ok(Some((mut frame, dur))) => {
-                    // at this point we have technical already activated
-                    // but we may want to scrap it if it is not going to end well
+                    deps_time += dur;
+
+                    // Our `frame` here is a new package with its own list of
+                    // dependencies. Do a sanity check here of all those
+                    // dependencies by cross-referencing our global
+                    // `past_conflicting_activations`. Recall that map is a
+                    // global cache which lists sets of packages where, when
+                    // activated, the dependency is unresolvable.
+                    //
+                    // If any our our frame's dependencies fit in that bucket,
+                    // aka known unresolvable, then we extend our own set of
+                    // conflicting activations with theirs. We can do this
+                    // because the set of conflicts we found implies the
+                    // dependency can't be activated which implies that we
+                    // ourselves can't be activated, so we know that they
+                    // conflict with us.
                     let mut has_past_conflicting_dep = just_here_for_the_error_messages;
                     if !has_past_conflicting_dep {
                         if let Some(conflicting) = frame
                             .remaining_siblings
                             .clone()
-                            .filter_map(|(_, (deb, _, _))| {
-                                past_conflicting_activations.get(&deb).and_then(|past_bad| {
```

Comment:
This part I was a [little confused about](https://github.com/rust-lang/cargo/pull/5168#issuecomment-373549968) given that we also modify the `past_conflicting_activations` above, and before merging I'd like to incorporate https://github.com/rust-lang/cargo/pull/5168#issuecomment-373596268 into the comment on this `if` statement